src/lib/services/feature-toggle-service.ts

```diff
@@ -314,51 +314,58 @@ class FeatureToggleService {
         const { featureName, projectId, environment } = context;
         await this.validateFeatureContext(context);
 
-        if (strategyConfig.constraints?.length > 0) {
-            strategyConfig.constraints = await this.validateConstraints(
-                strategyConfig.constraints,
-            );
-        }
+        if (!(await this.changeRequestsEnabled(projectId, environment))) {
```

Comment:
double nested if is usually a pattern of unnecessary complexity. Can we do early exit with the opposite condition?
if(changeRequestEnabled) throw Error